### PR TITLE
Copy the characters of a cesu-8 encoded substring into a specified buffer

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -1396,6 +1396,58 @@ jerry_string_to_utf8_char_buffer (const jerry_value_t value,
 - [jerry_create_string_from_utf8](#jerry_create_string_from_utf8)
 - [jerry_get_utf8_string_size](#jerry_get_utf8_string_size)
 
+## jerry_substring_to_char_buffer
+
+**Summary**
+
+Copy the characters of a cesu-8 encoded substring into a specified buffer.
+The '\0' character could occur in character buffer. Returns 0, if the value
+parameter is not a string. It will extract the parts of the string beetween
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md]: It will extract the substring between

File: tests/unit/test-api.c [hunk 2]
```diff
@@ -430,6 +430,71 @@ main (void)
...
+
+  sz = jerry_substring_to_char_buffer (args[0], 0, cesu8_length, (jerry_char_t *) supl_substring, cesu8_sz);
+  TEST_ASSERT (sz == 11);
+  TEST_ASSERT (!strncmp (supl_substring, "\x73\x74\x72\x3a \xed\xa0\x80\xed\xb6\x8a", sz));
+
+  /* Decrease the buffer size => the low surrogate char will not fit into the buffer */
+  cesu8_sz -= 1;
+  sz = jerry_substring_to_char_buffer (args[0], 0, cesu8_length, (jerry_char_t *) supl_substring, cesu8_sz);
+  TEST_ASSERT (sz == 8);
+  TEST_ASSERT (!strncmp (supl_substring, "\x73\x74\x72\x3a \xed\xa0\x80", sz));
```
Comment (zherczeg) [inline tests/unit/test-api.c]: There should be a test where offset > 0

Comment (robertsipka) [inline tests/unit/test-api.c]: Thanks, I've updated this patch.

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**